File: src/main/java/personal/ffivquizbot/slash_command/astrologian_skills/enums/DrawSkills.java
package personal.ffivquizbot.slash_command.astrologian_skills.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;


/**
 * @description 점지(Draw) 시 발동 가능한 스킬 목록
 */
@RequiredArgsConstructor
@Getter
public enum DrawSkills {
    BALANCE("아제마의 균형", "", DrawSkillTargets.DEFENSE_CLOSE, DrawSign.SUN),
    BOLE("세계수의 줄기", "", DrawSkillTargets.HEAL_FAR, DrawSign.SUN),
    ARROW("오쉬온의 화살", "", DrawSkillTargets.DEFENSE_CLOSE, DrawSign.MOON),
    SPEAR("할로네의 창", "", DrawSkillTargets.DEFENSE_CLOSE, DrawSign.STAR),
    EWER("살리아크의 물병", "", DrawSkillTargets.HEAL_FAR, DrawSign.MOON),
    SPIRE("비레고의 탑", "", DrawSkillTargets.HEAL_FAR, DrawSign.STAR);

    private final String skillName;
    private final String skillIconUrl;
    private final DrawSkillTargets targets;
    private final DrawSign sign;

}

File: src/main/kotlin/personal/ffivquizbot/slashcommand/astrologianskills/enums/DrawSkills.kt
package personal.ffivquizbot.slashcommand.astrologianskills.enums


enum class DrawSkills(
    val skillName:String,
    val skillIconUrlSuffix:String,
    val target:DrawSkillTargets,
    val sign:DrawSigns,
) {
    BALANCE("아제마의 균형", "Azema.png", DrawSkillTargets.DEFENSE_CLOSE, DrawSigns.SUN),
    BOLE("세계수의 줄기", "WorldTree.png", DrawSkillTargets.HEAL_FAR, DrawSigns.SUN),
    ARROW("오쉬온의 화살", "Oshion.png", DrawSkillTargets.DEFENSE_CLOSE, DrawSigns.MOON),
    SPEAR("할로네의 창", "Halone.png", DrawSkillTargets.DEFENSE_CLOSE, DrawSigns.STAR),
    EWER("살리아크의 물병", "Saliak.png", DrawSkillTargets.HEAL_FAR, DrawSigns.MOON),
     SPIRE("비레고의 탑", "Birego.png", DrawSkillTargets.HEAL_FAR, DrawSigns.STAR),
}

